src/centralized_telescope/algorithm.rs

```diff
@@ -52,6 +52,24 @@ pub fn verify(setup: &Setup, proof: &Proof) -> bool {
     proof_hash(setup, &round)
 }
 
+/// Alba's proving algorithm used for benchmarking, returning a boolean
+/// according to whether a proof has been found as well as the number of steps
+/// and reties done.
+pub fn bench(setup: &Setup, prover_set: &[Element]) -> (u64, u64, bool) {
```

Comment:
It would be nice to make a new function returning a proof and number of steps and call it from both `prove()` and `bench()` to avoid duplicating code and make sure we are testing the same thing.